docs/hbi-migration-runbook.md

```diff
@@ -0,0 +1,105 @@
+```
+# Add tables to HBI db
+./deploy.sh host-replication-tables
+```
+```
+# Add kessel kafka infra
+./deploy.sh host-replication-kafka
+```
+
+### Generate & Insert HBI records
+If you don't have host data already generated, generate it from the `db-generator` repository.
+
+The following generates 10 host files with 1000 host records each (hosts are counted in hundreds and multiplied by the number of files). This means the following command will generate 10,000 host records total.
+```
+go run main.go --num-hosts 10 --num-files 10 --type hosts 
+```
+
+Grab the host-inventory-db database `user` secret from the ephemeral environment.
+```
+oc get secrets host-inventory-db -o json | jq '.data["db.user"]' -r | base64 -d
+```
+
+Now port forward to the host-inventory-db database
+```
+oc port-forward service/host-inventory-db 5433:5432
+```
+
+Next, insert the HBI records into the ephemeral environment database.
+```
+psql -h localhost -p 5433 -U <user> -d host-inventory-db -f <path-to-generated-host-file>
+```
+
+You can validate the records are inserted by checking the `host-inventory-db` database:
+```
+psql -h localhost -p 5433 -U <user> -d host-inventory-db -c "select count(*) from hosts" -x
+```
+
+### Start the Migration
+
+To start the HBI migration, you just need to set the host migration connector to `running`
+
+```
+oc patch kafkaconnector "hbi-migration-connector" --type='merge' -p='{"spec":{"state":"running"}}'
+```
+
+### Validating the Migration
+
+To validate the migration, you can check the `kessel-inventory-consumer` pod logs for the following message:
+```
+INFO ts=2025-08-06T18:10:20Z caller=log/log.go:30 service.name=inventory-consumer service.version=0.1.0 trace.id= span.id= subsystem=inventoryConsumer msg=consumed event from topic host-inventory.hbi.hosts, partition 0 at offset 999
+```
+
+To validate database records, you can port-forward and check the `kessel-inventory` database for the following records:
+
+**Get the number of resources in the kessel-inventory database**
+```
+psql -h localhost -p <local_port> -U <user> -d kessel-inventory -c "select count(*) from resources" -x
+```
+
+**Check for resources in the kessel-inventory database that do not have a consistency token (should be 0)**
+```
+psql -h localhost -p <local_port> -U <user> -d kessel-inventory -c "select count(*) from resources where consistency_token != ''" -x
+```
+
+### Outbox Replication
+
+To validate outbox replication, you can do the same process with a couple of changes
+1. Set the outbox connector to `running` first
+```
+oc patch kafkaconnector "hbi-outbox-connector" --type='merge' -p='{"spec":{"state":"running"}}'
+```
+2. Generate outbox records via the `db-generator` repository using the `--type outbox` flag
+
+Once you make those changes, you can insert and validate the outbox records are flowing through the same means as the host migration.
+
+### Resetting the Migration
```

Comment:
FYI: https://github.com/project-kessel/insights-service-deployer/pull/59/files adds a clean up script so this could change to use that

./scripts/migration-cleanup.sh will nuke pretty much everything done with those 2 hbi commands

Reply:
Thanks! Added a small blurb on this.